### PR TITLE
CE-3460 Initialize the HTE demo when user scrolls to content

### PR DESCRIPTION
**Pull request by adamkarminski**
## Links
* http://wikia-inc.atlassian.net/browse/CE-3460

## Description
The demo will now be launched when a user scrolls to an article's content.

## Reviewers
@Wikia/community-engineering @Wikia/x-wing

### CODE REVIEW

File: front/main/app/components/article-content.js [hunk 1]
```diff
@@ -569,33 +572,84 @@ export default Ember.Component.extend(
...
-					range.selectNodeContents($highlightedElement[0]);
-					selection.removeAllRanges();
-					selection.addRange(range);
+					this.set('targetParagraphOffset', $paragraph.offset().top);
 
-					Ember.run.later(() => {
-						$highlightedElement.trigger('mousedown');
-					}, 500);
+					Ember.$(document).on('touchmove', this, this.debouncedScroll);
+					Ember.$(window).on('scroll', this, this.debouncedScroll);
```
Comment (lukaszkonieczny) [inline front/main/app/components/article-content.js]: I think these events should be bind if there is a word longer than `minLettersLimit`

Comment (lukaszkonieczny) [inline front/main/app/components/article-content.js]: Also you can consider using `one` method which will trigger just once.

File: front/main/app/components/article-content.js [hunk 2]
```diff
@@ -569,33 +572,84 @@ export default Ember.Component.extend(
...
+
+			Ember.$(document).unbind('touchmove', this.debouncedScroll);
+			Ember.$(window).unbind('scroll', this.debouncedScroll);
+
+			if ($highlightedElement) {
+				range.selectNodeContents($highlightedElement[0]);
+				selection.removeAllRanges();
+				selection.addRange(range);
+
+				Ember.$('html, body').animate({scrollTop: `${$highlightedElement.offset().top - 150}px`});
```
Comment (lukaszkonieczny) [inline front/main/app/components/article-content.js]: Is it some kind of workaround for something or we want to animate both elements?

Comment (kamilkoterba) [inline front/main/app/components/article-content.js]: http://stackoverflow.com/questions/19303405/difference-between-html-body-animate-and-body-animate
Both sounds safer but possibly this browser differences are not longer valid. We can test this if we have spare time before shipping the experiment.

**Comment by adamkarminski** [comment]
LGTM!

**Pull request merged**